docs/howto/csrf-protection.txt

```diff
@@ -0,0 +1,321 @@
+
+The error page, however, is not very friendly, so you may want to provide your
+own view for handling this condition.  To do this, simply set the
+:setting:`CSRF_FAILURE_VIEW` setting.
+
+CSRF failures are logged as warnings to the :ref:`django.security.csrf
+<django-security-logger>` logger.
+
+.. versionchanged:: 1.11
+
+    In older versions, CSRF failures are logged to the ``django.request``
+    logger.
+
+Using CSRF protection with caching
+==================================
+
+If the :ttag:`csrf_token` template tag is used by a template (or the
+``get_token`` function is called some other way), ``CsrfViewMiddleware`` will
+add a cookie and a ``Vary: Cookie`` header to the response. This means that the
+middleware will play well with the cache middleware if it is used as instructed
+(``UpdateCacheMiddleware`` goes before all other middleware).
+
+However, if you use cache decorators on individual views, the CSRF middleware
+will not yet have been able to set the Vary header or the CSRF cookie, and the
+response will be cached without either one. In this case, on any views that
+will require a CSRF token to be inserted you should use the
+:func:`django.views.decorators.csrf.csrf_protect` decorator first::
+
+  from django.views.decorators.cache import cache_page
+  from django.views.decorators.csrf import csrf_protect
+
+  @cache_page(60 * 15)
+  @csrf_protect
+  def my_view(request):
+      ...
+
+If you are using class-based views, you can refer to :ref:`Decorating
+class-based views<decorating-class-based-views>`.
+
+Testing and CSRF protection
+===========================
+
+The ``CsrfViewMiddleware`` will usually be a big hindrance to testing view
+functions, due to the need for the CSRF token which must be sent with every POST
+request.  For this reason, Django's HTTP client for tests has been modified to
+set a flag on requests which relaxes the middleware and the ``csrf_protect``
+decorator so that they no longer rejects requests.  In every other respect
+(e.g. sending cookies etc.), they behave the same.
+
+If, for some reason, you *want* the test client to perform CSRF
+checks, you can create an instance of the test client that enforces
+CSRF checks::
+
+    >>> from django.test import Client
+    >>> csrf_client = Client(enforce_csrf_checks=True)
+
+.. _csrf-edge-cases:
+
+Edge cases
+==========
+
+Certain views can have unusual requirements that mean they don't fit the normal
+pattern envisaged here. A number of :ref:`CSRF protection utilities
+<csrf-protection-utilities>` can be useful in these scenarios:
+
+Disabling CSRF protection for just a few views
+----------------------------------------------
+
+Most views requires CSRF protection, but a few do not.
+
+Solution: rather than disabling the middleware and applying ``csrf_protect`` to
+all the views that need it, enable the middleware and use
+:func:`~django.views.decorators.csrf.csrf_exempt`.
+
+Setting the token when CsrfViewMiddleware.process_view is not used
```

Comment:
Revisit that paragraph, as of https://github.com/django/django/commit/c4c128d67c7dc2830631c6859a204c9d259f1fb1 this is probably no longer true.

Reply:
Hah, then https://github.com/django/django/commit/c4c128d67c7dc2830631c6859a204c9d259f1fb1 should have targeted these docs too!

Reply:
Yes :þ